Peer review

aimeedyki opened the pull request.

aimeedyki committed:
hashes password

aimeedyki committed:
Merge pull request #1 from aimeedyki/signup

hashes password

aimeedyki committed:
Merge pull request #2 from aimeedyki/signin

creates initial sign in

aimeedyki committed:
Merge pull request #3 from aimeedyki/develop

Develop

aimeedyki committed:
Merge pull request #4 from aimeedyki/addbook

Addbook

aimeedyki committed:
Merge pull request #6 from aimeedyki/displaybooks

Displaybooks

aimeedyki committed:
Merge pull request #7 from aimeedyki/borrow

creates route to borrow a book

aimeedyki committed:
Merge pull request #8 from aimeedyki/category

creates and displays categories

aimeedyki committed:
Merge pull request #9 from aimeedyki/feature/validated-signup

adds validation to user signup feature

aimeedyki committed:
Merge pull request #10 from aimeedyki/feature/add-book

creates route to add a book

aimeedyki committed:
Merge pull request #11 from aimeedyki/feature/modify-book-route

creates route to modify and display books

aimeedyki committed:
refactors code to ES6 format

aimeedyki committed:
Merge pull request #13 from aimeedyki/refactor

Refactor

aimeedyki committed:
creates routes to update and add a book with validation

aimeedyki committed:
Merge pull request #14 from aimeedyki/feature/validated-book-routes

creates routes to update and add a book with validation

aimeedyki committed:
modifies login to include jwt

aimeedyki committed:
Merge pull request #15 from aimeedyki/feature/jwt-login

modifies login to include jwt

aimeedyki committed:
writes tests for routes

aimeedyki committed:
updates routes to pass tests

aimeedyki committed:
sets up code checking servces

aimeedyki committed:
Merge pull request #17 from aimeedyki/test-setup

Test setup

aimeedyki committed:
adds procfile for heroku

aimeedyki committed:
Merge pull request #18 from aimeedyki/test-setup

adds procfile for heroku

aimeedyki committed:
removes node-mon from start script for heroku

aimeedyki committed:
Merge pull request #19 from aimeedyki/test-setup

removes node-mon from start script for heroku

aimeedyki committed:
changes commands in procfile for heroku

aimeedyki committed:
changes commands in procfile for heroku

aimeedyki committed:
Merge pull request #20 from aimeedyki/test-setup

changes commands in procfile for heroku

aimeedyki committed:
Merge branch 'test-setup' of https://github.com/aimeedyki/hello-Books into test-setup

aimeedyki committed:
ammends package.jos for heroku

aimeedyki committed:
Merge pull request #21 from aimeedyki/test-setup

Test setup

aimeedyki committed:
changes package.json

aimeedyki committed:
adds .babelrc

aimeedyki committed:
inserts appropraite comments

aimeedyki committed:
refactor code to handle additional route validations that arose from tests

aimeedyki committed:
Merge pull request #22 from aimeedyki/test-setup

changes package.json

aimeedyki committed:
add authorization feature

aimeedyki committed:
Merge pull request #23 from aimeedyki/ft-view-books-endpoint-149610555

add authorization feature

aimeedyki committed:
add heroku post buid script

aimeedyki committed:
modify config.json

aimeedyki committed:
undo postgres github commands

aimeedyki committed:
migrate to elephant sql

aimeedyki committed:
Create README.md

aimeedyki committed:
create endpoint to display user's profile

aimeedyki committed:
create endpoint for displaying books by categories

aimeedyki committed:
create route to delete a book

aimeedyki committed:
Merge pull request #24 from aimeedyki/ft-delete-book-endpoint-150255518

Ft delete book endpoint 150255518

aimeedyki committed:
create a route for admin to receive notifications

aimeedyki committed:
set different return dates and maximum books a user can borrow by levels

aimeedyki committed:
modify route to user levels

aimeedyki committed:
create route for a user to change password

aimeedyki committed:
Merge pull request #25 from aimeedyki/ft-change-password-150311477

Ft change password 150311477

aimeedyki committed:
write correct tests

aimeedyki committed:
Merge pull request #26 from aimeedyki/test-corrected

write correct tests

aimeedyki committed:
correct travis build

aimeedyki committed:
try to correct travis build

aimeedyki committed:
modify test script

aimeedyki committed:
set environment variables in travis

aimeedyki committed:
change travis setup

aimeedyki committed:
tweak travis setup

aimeedyki committed:
add sequelizerc file

aimeedyki committed:
log error message on travis build

aimeedyki committed:
log error message

aimeedyki committed:
write new setup for travis

aimeedyki committed:
correct line in package.json

aimeedyki committed:
remove set from test command in package.json

aimeedyki committed:
remove env test in package.json

aimeedyki committed:
log error from controller

aimeedyki committed:
modify package.json

aimeedyki committed:
modify models index

aimeedyki committed:
change models index

aimeedyki committed:
change package.json

aimeedyki committed:
correct package.json

aimeedyki committed:
correct package.json

aimeedyki committed:
correct package.json

aimeedyki committed:
clean code

aimeedyki committed:
change to travis database

aimeedyki committed:
modify test to pass travis

aimeedyki committed:
Merge pull request #27 from aimeedyki/ft-view-books-endpoint-149610555

Create README.md

aimeedyki committed:
Merge pull request #28 from aimeedyki/test

Test

aimeedyki committed:
Update README.md

aimeedyki committed:
Update README.md

aimeedyki committed:
set up swagger-ui and document signup route

aimeedyki committed:
Merge pull request #29 from aimeedyki/documentation

set up swagger-ui and document signup route

aimeedyki committed:
document all user routes

aimeedyki committed:
document all book routes

aimeedyki committed:
document all routes

aimeedyki committed:
set up react app

aimeedyki committed:
trobleshoot travis issue

aimeedyki committed:
trobleshoot travis issue

aimeedyki committed:
creates basic layout of user page

aimeedyki committed:
make tab component flexible

aimeedyki committed:
write all components

aimeedyki committed:
set up routing for all pages

aimeedyki committed:
set up redux with login and signup components

Commit:
New file commit for usage

aimeedyki committed:
create a login actions

Commit:
add gitignore

Commit:
create login user interface

aimeedyki committed:
Merge pull request #30 from aimeedyki/ft-user-login-ui-149610399

#149610399 Builds User Login and  Signup User Interaction

aimeedyki committed:
create a workable ui of user's page features

aimeedyki committed:
set up front end server from backend

aimeedyki committed:
create user signup feature

aimeedyki committed:
add Login feature

aimeedyki committed:
add login feature with route authentication

aimeedyki committed:
add feature to add a book to library

aimeedyki committed:
add feature for a user to view all books in a library

aimeedyki committed:
add feature that allows admin delete a book

aimeedyki committed:
add feature to allow a user borrow a book

aimeedyki committed:
add feature for user to view user profile"

aimeedyki committed:
refactor code

aimeedyki committed:
pass travis test

aimeedyki committed:
pass hound

aimeedyki committed:
pass hound violations

aimeedyki committed:
pass hound

aimeedyki committed:
resolve hound

aimeedyki committed:
resolve hound

aimeedyki committed:
resolve hound

aimeedyki committed:
resolve hound

aimeedyki committed:
resolve hound

aimeedyki committed:
resolve hound

aimeedyki committed:
resolve hound

aimeedyki committed:
resolve hound

aimeedyki committed:
resolve hound

aimeedyki committed:
resolve hound

aimeedyki committed:
resolve hound

aimeedyki committed:
resolve hound

aimeedyki committed:
Merge pull request #31 from aimeedyki/feature/149610767/view-user-profile

#149610767 view user profile

aimeedyki committed:
reconfigure coveralls

aimeedyki committed:
change npm cover to npm run cover

aimeedyki committed:
correct hound changes

aimeedyki committed:
resolve hound issues

aimeedyki committed:
Merge pull request #33 from aimeedyki/feature/149610767/view-user-profile

Feature/149610767/view user profile

aimeedyki committed:
change styling

aimeedyki committed:
change styling

aimeedyki committed:
add loaders to dependencies

aimeedyki committed:
Merge pull request #34 from aimeedyki/feature/149610767/view-user-profile

#149610767 view user profile

aimeedyki committed:
change app.js location

aimeedyki committed:
Merge pull request #35 from aimeedyki/feature/149610767/view-user-profile

change app.js location

aimeedyki committed:
create webpack for production

aimeedyki committed:
Merge pull request #36 from aimeedyki/feature/149610767/view-user-profile

create webpack for production

aimeedyki committed:
add picture component

aimeedyki committed:
Merge pull request #37 from aimeedyki/feature/149610767/view-user-profile

add picture component

aimeedyki committed:
feat(front-end): view user profile
- allow a user to see their profile

aimeedyki committed:
implement coveralls badge

aimeedyki committed:
Update README.md

aimeedyki committed:
add .eslintignore

aimeedyki committed:
Merge pull request #38 from aimeedyki/feature/149610767/view-user-profile

#149610767 view user profile

aimeedyki committed:
refactor: refactor code to meet best practices
- correct server responses and code
- implement feedback on front end

aimeedyki committed:
pass hound voilations

aimeedyki committed:
Merge pull request #39 from aimeedyki/code-refactor

refactor: refactor code to meet best practices

aimeedyki committed:
remove .ds_store from repo

aimeedyki committed:
solve hound violations

aimeedyki committed:
correct hound violations

aimeedyki committed:
correct hound violations

change alerts to sweet alert

enahomurphy commented on client/src/actions/authAction.js:
how valid is this logic, you're basically login out the user if there is forbidden error or a bad request. 

enahomurphy commented on client/src/actions/authAction.js:
Don't you think this should be in your env file or you can dynamically generate this.

enahomurphy commented on client/src/actions/bookAction.js:
your docs should follow JSDOC standard

enahomurphy commented on client/src/actions/types.js:
This is a good approach but I still feel this should go into your environment

enahomurphy commented on client/src/components/Library/Allbooks.jsx:
What is this method for?

enahomurphy commented on client/src/components/Profile/Profile.jsx:
Remove commented code if not needed

aimeedyki commented on client/src/components/Library/Allbooks.jsx:
it returns an array of the page numbers for the pagination component.

aimeedyki commented on client/src/actions/authAction.js:
Thanks, I have implemented this

aimeedyki commented on client/src/actions/bookAction.js:
Thanks, I will change them

aimeedyki commented on client/src/actions/types.js:
Thanks, I have implemented this

aimeedyki commented on client/src/components/Profile/Profile.jsx:
Thanks, I have implemented this

aimeedyki committed:
bugfix(pagination):
Seperate pagination into a component.

fix hound violations

aimeedyki committed:
Merge pull request #41 from aimeedyki/fix/pagination

Fix/pagination

aimeedyki committed:
chore(deployment) : deploy app to heroku

fix api url

change api server in app.js

change webpack prod

increase test

use webpackmiddleware for development only

define public path

change app.js

change cloudinary url

add url loader to webpack

add define plugin

add dev-server

change admin conditions

correct book return error

resolve home image path

change image url

aimeedyki committed:
Update README.md

aimeedyki committed:
Merge pull request #42 from aimeedyki/chore/152379230/deploy-app-to-heroku

#152379230 deploy app to heroku

aimeedyki committed:
feat(authentication):
Protect front-end routes
- ensure unauthenticated users can not access our routes
- ensure regular users can not access admin routes
[finishes #152450001]

rename auth component

increase test timeout

aimeedyki committed:
Merge pull request #43 from aimeedyki/feature/#152450001/protect-front-end-routes

#152450001 protect front end routes

aimeedyki committed:
fix(feedback): implement feedback on work

correct test

modify codeclimate

modify codeclimate

modify codeclimate

aimeedyki committed:
Merge pull request #45 from aimeedyki/fix/feedback-implementation

Feedback implementation

aimeedyki committed:
feat(search): allow users to search for books in the library

solve hound issues

aimeedyki committed:
Merge pull request #46 from aimeedyki/feature/152923095/book-search

#152923095 book search

aimeedyki committed:
feat(category): modify and edit categories
- allow admin users to modify and delete caategories

aimeedyki committed:
Merge pull request #47 from aimeedyki/feature/152992085/delete-modify-category

#152992085 modify and edit categories

aimeedyki committed:
fix resolve

solve file name change disparity

further file changes

remove log statements

correct category error

solve search issues when searching categories

fix image loader

insert close buttons

add pagination to test

change seed file

aimeedyki committed:
Merge pull request #48 from aimeedyki/bug-fix/heroku-re-deployment

fix heroku app deployment

aimeedyki committed:
feat(authentication): add google authentication

correct function length

aimeedyki committed:
Merge pull request #49 from aimeedyki/feature/150161851/google-authentication

#150161851 google authentication

aimeedyki committed:
feat(surcharge): send mail on surcharge to defaulting customers

hide environment variables

change details

aimeedyki committed:
Merge pull request #50 from aimeedyki/feature/149611913/email-on-surcharge

#149611913 send mail on surcharge

aimeedyki committed:
feat(transactions): allow users to upload surcharge and subscription payment  and admin confirms it

fix test

aimeedyki committed:
Merge pull request #51 from aimeedyki/feature/153218009/surcharge-subscription-confirmation

#153218009 upload and confirm transactions

aimeedyki committed:
chore(test): setup and carry out front end tests

write more client side tests

write action and recucer tests

upgrade swagger ui

set up end to end test

update readme

change api docs url

change redirect url

fix test issues

fix footer

add test db script for travis

change babelrc

aimeedyki committed:
Merge pull request #52 from aimeedyki/chore/153261048/setup-front-end-testing

#153261048 front end tests

aimeedyki committed:
fix(feedback): implement feedback from review
- fix footer to bottom
- remove unnecessary lines of code
- make filenames match default exports

add jsdocs and give appropraite file names

clear search errors in searchBook action

rename test files

style admin dashboard

fix pagination

update readme

refactor tests to be more specific and descriptive

remove trailing spaces

aimeedyki committed:
Merge pull request #54 from aimeedyki/chore/scorecard-feedback-implementation

Implement feedback from review
